homeassistant/components/zwave/node_entity.py

```diff
@@ -74,6 +75,15 @@ def try_remove_and_add(self):
         if self.hass and self.platform:
             self.hass.add_job(_async_remove_and_add)
 
+    async def node_removed(self):
+        """Call when a node is removed from the Z-Wave network."""
+        registry = await async_get_registry(self.hass)
+        if self.entity_id not in registry.entities:
+            return
```

Comment:
Why would you not call `self.async_remove()` if it's not in a registry?

Reply:
Good catch, that wasn't intentional. Wasn't paying attention to the fact the guard clause would stop execution before it got to that step.